pkg/operator/deploy/deploy.go

```diff
@@ -138,6 +138,15 @@ func (o *operator) resources() ([]kruntime.Object, error) {
 		domain += "." + o.env.Domain()
 	}
 
+	ingressIP := o.oc.Properties.IngressProfiles[0].IP
+	if len(o.oc.Properties.IngressProfiles) > 1 {
+		for _, p := range o.oc.Properties.IngressProfiles {
+			if p.Name == "default" {
+				ingressIP = p.IP
+			}
```

Comment:
Ingress profile with name `"default"` is not guaranteed to exist. I think customers can create an ingress controller `"foo"` and delete `"default"`. What happens in this case?

Reply:
@karanmagdani can we combine my suggestion of adding some unit testing here with @m1kola's points about edge cases and make sure we are properly handling all cases?

Reply:
Added test cases and covered in function. 
For case when: there is no default, it just picks the first ingress profile available (what we currently do). IMO this is fine because i don't think we have a way to find the "non-default" default